packages/block-editor/src/components/iframe/index.js

```diff
@@ -65,6 +64,14 @@ function useStylesCompatibility() {
 				return;
 			}
 
+			let isFilenameMatch = false;
+			if ( styleSheet.href ) {
+				const url = new URL( styleSheet.href );
+				if ( url.pathname === '/wp-admin/css/common.css' ) {
```

Comment:
Is it okay to check for the full pathname? Is the check sufficient?

Do we need an allowlist of other Core styles that we want to carry over to the iframe?

Reply:
why do we need this stylesheet inside the iframe? AFAIK it's not something that targets frontend?

AFAIK `alignleft` and `alignright` are either supposed to be provided by the theme for classic themes or injected in layout styles for block themes.

Is the issue here about classic or block themes?

Reply:
> why do we need this stylesheet inside the iframe? AFAIK it's not something that targets frontend?

AFAICT, it is being used on the frontend by Core's `theme-compat/comments.php` fallback, which in turn is used by the Post Comments block (which is a thin wrapper around `comments_template()`):

https://github.com/WordPress/wordpress-develop/blob/b27069117eb22c39d5230ceb39439f2dab82dc08/src/wp-includes/theme-compat/comments.php#L52-L64

> Is the issue here about classic or block themes?

Block themes -- I think that's the main target for the Post Comments block.

Reply:
> AFAICT, it is being used on the frontend by Core's theme-compat/comments.php fallback, which in turn is used by the Post Comments block (which is a thin wrapper around comments_template()):

I guess that also means that the current fix is not enough since it only impacts the editor. I think the best solution here is to just add this to the "style.css" of the block like suggested by @Mamaduka that way it applies in both editor and frontend regardless of the theme used. Also, you might you want to target these links specifically and avoid a global rule that apply cross blocks and markup.

Reply:
>I guess that also means that the current fix is not enough since it only impacts the editor. I think the best solution here is to just add this to the "style.css" of the block like suggested by @Mamaduka that way it applies in both editor and frontend regardless of the theme used. 

Thanks! I had some [reservations](https://github.com/WordPress/gutenberg/pull/40846#issuecomment-1118635608) about this, but I guess it might be the most pragmatic approach then 😅 

> Also, you might you want to target these links specifically and avoid a global rule that apply cross blocks and markup.

Can do. Still feels like there might be some need for a more far-reaching solution that makes some of these classes available that are relied on by Core functions that are used by some dynamic blocks on the frontend, but maybe we don't have to solve that here and now... 😬

Reply:
>  Still feels like there might be some need for a more far-reaching solution that makes some of these classes available that are relied on by Core functions that are used by some dynamic blocks on the frontend, but maybe we don't have to solve that here and now...

No, I actually think the opposite. The alignment styles are not generated by layout in block themes for all use-cases. The issue here is an outlier and that doesn't deserve a global fix. In other words blocks shouldn't be assuming that they can use "alignleft", "alignright" and expect these to be defined in all situations. These define on their container for block themes.

So actually, the block could have use custom classes and just target these instead but since it's an old template from Core. We can just add specific styles to the block using the old template.